Extensions/Share/ShareViewController.swift

```diff
@@ -0,0 +1,306 @@
+        return view
+    }
+
+    private func layoutSeparators() {
+        separators.forEach {
+            $0.snp.makeConstraints { make in
+                make.left.right.equalToSuperview()
+                make.height.equalTo(1)
+            }
+        }
+    }
+
+    private func makePageInfoRow() -> (row: UIView, pageTitleLabel: UILabel, urlLabel: UILabel) {
+        let row = UIView()
+        let pageTitleLabel = UILabel()
+        let urlLabel = UILabel()
+
+        [pageTitleLabel, urlLabel].forEach { label in
+            row.addSubview(label)
+            label.allowsDefaultTighteningForTruncation = true
+            label.lineBreakMode = .byTruncatingMiddle
+            label.font = UX.baseFont
+        }
+
+        pageTitleLabel.font = UIFont.boldSystemFont(ofSize: UX.baseFont.pointSize)
+        pageTitleLabel.snp.makeConstraints { make in
+            make.right.equalToSuperview().inset(UX.rowInset)
+            make.left.equalToSuperview().inset(UX.pageInfoRowLeftInset)
+            make.bottom.equalTo(row.snp.centerY)
+        }
+
+        urlLabel.snp.makeConstraints {
+            make in
+            make.right.equalToSuperview().inset(UX.rowInset)
+            make.left.equalToSuperview().inset(UX.pageInfoRowLeftInset)
+            make.top.equalTo(pageTitleLabel.snp.bottom).offset(4)
+        }
+
+        return (row, pageTitleLabel, urlLabel)
+    }
+
+    private func makeActionRow(label: String, imageName: String, action: @escaping (() -> Void), hasNavigation: Bool) -> UIView {
+        let row = UIView()
+        let icon = UIImageView(image: UIImage(named: imageName))
+        icon.contentMode = .scaleAspectFit
+
+        let title = UILabel()
+        title.font = UX.baseFont
+
+        title.text = label
+        [icon, title].forEach { row.addSubview($0) }
+
+        icon.snp.makeConstraints { make in
+            make.left.equalToSuperview().inset(UX.rowInset)
+            make.centerY.equalToSuperview()
+            make.width.equalTo(34)
+        }
+
+        title.snp.makeConstraints { make in
+            make.right.equalToSuperview()
+            make.centerY.equalToSuperview()
+            make.left.equalTo(icon.snp.right).offset(UX.actionRowSpacingBetweenIconAndTitle)
+        }
+
+        if hasNavigation {
+            let navButton = UIImageView(image: UIImage(named: "menu-Disclosure"))
+            navButton.contentMode = .scaleAspectFit
+            row.addSubview(navButton)
+            navButton.snp.makeConstraints { make in
+                make.right.equalToSuperview().inset(UX.rowInset)
+                make.centerY.equalToSuperview()
+                make.width.height.equalTo(14)
+            }
+        }
+
+        let gesture = UITapGestureRecognizer(target: self, action:  #selector(handleRowTapGesture))
+        row.addGestureRecognizer(gesture)
+        // map the gesture to the action func that will be called when the gesture is performed
+        actions[gesture] = action
```

Comment:
Instead of keeping a ref to all the gestures. We could pass in a selector to `makeActionRow` of the function that should be called when tapped.

Reply:
headsmack! of course, fixing.